Lib/glyphsLib/glyphdata.py

```diff
@@ -21,21 +21,126 @@
 """
 
 
-import collections
 import re
+import os
 from fontTools import unicodedata
+
 import xml.etree.ElementTree
 
 import fontTools.agl
 
 
-__all__ = ["get_glyph", "GlyphData"]
+__all__ = [
+    "get_glyph",
+    "GlyphData",
+    "GlyphInfo",
+    "GSUppercase",
+    "GSLowercase",
+    "GSSmallcaps",
+    "GSMinor",
+]
+
+GSNoCase = None  # 0
+GSUppercase = "upper"  # 1
+GSLowercase = "lower"  # 2
+GSSmallcaps = "small"  # 3
+GSMinor = "minor"  # 4
+
+GSBIDI = "BIDI"
+GSLTR = "LTR"
+GSRTL = "RTL"
+GSVertical = 4
+
+
+def debug(*string):
+    # print(*string)
+    pass
+
+
+class GlyphInfo:
```

Comment:
Given that we use Python 3.7 as the minimum, GlyphsInfo can be made a https://devdocs.io/python~3.7/library/dataclasses, which saves some boilerplate code.

Reply:
I tried that when I was working on the glyphData stuff. But I didn’t figure out how to implement the fallback of the production name property. And because of that property, the custom copy() method is still needed and so it saves not as much as it could.